src/GUI/EFCorePowerTools/ViewModels/PickServerDatabaseViewModel.cs

```diff
@@ -109,5 +120,24 @@ private void Cancel_Executed()
             SelectedDatabaseDefinition = null;
             CloseRequested?.Invoke(this, new CloseRequestedEventArgs(false));
         }
+
+        private DatabaseDefinitionModel PreSelectDatabaseDefinition()
+        {
+            var subset = DatabaseDefinitions.Where(m =>
+                                             {
+                                                 if (!m.FilePath.EndsWith(".sqlproj"))
+                                                 {
+                                                     return false;
+                                                 }
+
+                                                 var fileName = Path.GetFileNameWithoutExtension(m.FilePath).ToLower();
+                                                 return !fileName.StartsWith("test")
+                                                     && !fileName.EndsWith("test");
```

Comment:
This seems brittle?

Reply:
To what extend?
There's no better way to find "test" projects, rather than by naming convention. 🤔

Reply:
Are these test database projects? (sqlproj) ?

Reply:
I think this must be removed, there is only .sqlproj

Reply:
As said, it's by naming convention, not by project type or file extension.

e.g. you have these two database projects:

- BookingDatabase.sqlproj
- BookingDatabase.Test.sqlproj

`BookingDatabase.sqlproj` is the primary database project, defining the schema.
`BookingDatabase.Test.sqlproj` references `BookingDatabase.sqlproj`, and is used for test-deployments (e.g. builds) of the main project, seeding data, etc etc.

That's what the filter is for. And it doesn't filter the available projects in the dropdown, it just filters for the pre-selection. So if you have those two projects from the example in your solution, it'll pre-select the primary database project, rather than a project that doesn't define any objects in itself.

Reply:
I still think this is too localized and not a standard convention.